examples/regression/script.jl

```diff
@@ -0,0 +1,196 @@
+
+# Create the 'model' from the parameters - i.e. return the FiniteGP at inputs x,
+# the FiniteGP at inducing inputs z and the variational posterior over inducing
+# points - q(u).
+
+lik_noise = 0.3
+jitter = 1e-5
+
+# Next, we define some useful functions on the model - creating the prior GP
+# under the model, as well as the `SVGP` struct needed to create the posterior
+# approximation and to compute the ELBO.
+
+function prior(m::SVGPModel)
+    kernel = make_kernel(m.k)
+    return GP(kernel)
+end
+
+function make_approx(m::SVGPModel, prior)
+    q = MvNormal(m.m, m.A'm.A)
+    fz = prior(m.z, jitter)
+    return SVGP(fz, q)
+end
+#md nothing #hide
+
+# Create the approximate posterior GP under the model.
+
+function model_posterior(m::SVGPModel)
+    svgp = make_approx(m, prior(m))
+    return posterior(svgp)
+end
+#md nothing #hide
+
+# Define a predictive function for the model - in this case the prediction is
+# the joint distribution of the approximate posterior GP at some test inputs `x`
+# (defined by an `AbstractGPs.FiniteGP`).
+
+function (m::SVGPModel)(x)
+    post = model_posterior(m)
+    return post(x)
+end
+#md nothing #hide
+
+# Return the loss given data - for the SVGP, the loss used is the negative ELBO
+# (also known as the Variational Free Energy). `n_data` is required for
+# minibatching used below.
+
+function loss(m::SVGPModel, x, y; n_data=length(y))
+    f = prior(m)
+    fx = f(x, lik_noise)
+    svgp = make_approx(m, f)
+    return -elbo(svgp, fx, y; n_data)
+end
+#md nothing #hide
+
+# Finally, create the model with initial parameters
+
+model = SVGPModel(k_init, z_init, m_init, A_init)
+#md nothing #hide
+
+# ## Training the model
+#
+# Training the model now simply proceeds with the usual `Flux.jl` training loop.
+
+opt = ADAM(0.001)  # Define the optimiser
+params = Flux.params(model)  # Extract the model parameters
+#md nothing #hide
+
+# One of the major advantages of the SVGP model is that it allows stochastic
+# estimation of the ELBO by using minibatching of the training data. This is
+# very straightforward to achieve with `Flux.jl`'s utilities:
+
+b = 100 # minibatch size
+data_loader = Flux.Data.DataLoader((x, y); batchsize=b)
+
+# The loss (negative ELBO) before training
+
+println(loss(model, x, y))
+
+# Train the model
+
+Flux.train!(
+    (x, y) -> loss(model, x, y; n_data=N),
+    params,
+    ncycle(data_loader, 300), # Train for 300 epochs
+    opt,
+)
+#md nothing #hide
+
+# Negative ELBO after training
+
+println(loss(model, x, y))
+
+# Finally, we plot samples from the optimised approximate posterior to see the
+# results.
+
+post = model_posterior(model)
+
+scatter(
+    x,
+    y;
+    markershape=:xcross,
+    markeralpha=0.1,
+    xlim=(-1, 1),
+    xlabel="x",
+    ylabel="y",
+    title="posterior (VI with sparse grid)",
+    label="Train Data",
+)
+plot!(-1:0.001:1, post; label="Posterior")
+plot!(-1:0.001:1, g; label="True Function")
+vline!(z_init; label="Pseudo-points")
```

Comment:
Could you change it to only mark the location as little ticks `|` on the x-axis (if you can get Plots.jl to let you overplot "outside the axes")? it looks quite cluttered at the moment